rfd/0178-github-proxy.md

```diff
@@ -0,0 +1,497 @@
+---
+author: Steve Huang (xin.huang@goteleport.com)
+state: draft
+---
+
+# RFD 178 - GitHub Proxy
+
+## Required Approvers
+
+- Engineering: @r0mant && @smallinsky
+- Product: @klizhentas || @xinding33
+
+## What
+
+This RFD proposes design and implementation of proxying the Git SSH protocol
+for GitHub repositories.
+
+## Why
+
+GitHub Enterprise provides a security feature to bring your own SSH certificate
+authorities (CA). Once a CA is added, your organization can sign short-lived
+client SSH certificates to access organization resources on GitHub. You can
+also require your memebers to use these SSH certificates, which disables Git
+access using personal tokens.
+
+The concept of short-lived SSH certificates to access organization resources
+aligns well with Teleport, where a Teleport user begins their day with a 'tsh'
+session, accessing only what their roleset permits. Teleport can also easily
+provide the capability to issue of short-lived client SSH certificates for
+GitHub organzations so Teleport customers do not need to implement a separate
+system for issuing these certificates. 
+
+Teleport also offers other GitHub-related features, such as [GitHub IAM
+integration](https://github.com/gravitational/teleport.e/blob/master/rfd/0021e-github-iam-integration.md)
+and GitHub SSO, where this functionality can integrate nicely.
```

Comment:
Teleport already provides the "Using Teleport's Certificate Authority with GitHub" [integration](https://goteleport.com/docs/admin-guides/management/guides/ssh-key-extensions/).

I would mention other benefits of the GitHub Proxy, such as auditing GitHub access. Adding Support  MFA for critical GitHub actions.

Reply:
didn't know this. we can close the ticket now =p.

> I would mention other benefits of the GitHub Proxy, such as auditing GitHub access. Adding Support MFA for critical GitHub actions.

Ya, let me go through the existing integration and update the RFD. Thanks!